clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/ops/UpsertEntityTagsAtomicOperation.java

```diff
@@ -51,28 +54,15 @@ public UpsertEntityTagsAtomicOperation(Front50Service front50Service,
   }
 
   public Void operate(List priorOutputs) {
-    EntityTags.EntityRef entityRef = entityTagsDescription.getEntityRef();
-    String entityRefAccount = (String) entityRef.attributes().get("account");
-    String entityRefAccountId = (String) entityRef.attributes().get("accountId");
-    if (entityRefAccount != null && entityRefAccountId == null) {
-      // add `accountId` if available (and not already specified)
-      AccountCredentials accountCredentials = accountCredentialsProvider.getCredentials(entityRefAccount);
-      if (accountCredentials != null) {
-        entityRefAccountId = accountCredentials.getAccountId();
-        entityRef.attributes().put("accountId", entityRefAccountId);
-      }
+    if (entityTagsDescription.getId() == null) {
+      setIdAndIdPattern(entityTagsDescription);
```

Comment:
`entityTagsDescription` is marked as `final` - does it make sense to change that?

I don't think removing `final` will cause any underlying issues, but assigning the return value is a little misleading, as it implies (to me, at least) that the `setIdAndIdPattern` method returns a _different_ object than the one passed in, when it just performs some mutation.

I feel like I'm missing some pattern/practical context here.